src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -249,79 +315,467 @@ public DateTime(int year, int month, int day, int hour, int minute, int second,
             }
         }
 
-        // Constructs a DateTime from a given year, month, day, hour,
-        // minute, and second.
-        //
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DateTime"/> structure to the specified year, month, day, hour, minute, second,
+        /// millisecond, and Coordinated Universal Time (UTC) or local time for the specified calendar.
+        /// </summary>
+        /// <param name="year">The year (1 through 9999).</param>
+        /// <param name="month">The month (1 through 12).</param>
+        /// <param name="day">The day (1 through the number of days in <paramref name="month"/>).</param>
+        /// <param name="hour">The hours (0 through 23).</param>
+        /// <param name="minute">The minutes (0 through 59).</param>
+        /// <param name="second">The seconds (0 through 59).</param>
+        /// <param name="millisecond">The milliseconds (0 through 999).</param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="year"/> is less than 1 or greater than 9999.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="month"/> is less than 1 or greater than 12.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="day"/> is less than 1 or greater than the number of days in <paramref name="month"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="hour"/> is less than 0 or greater than 23.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="minute"/> is less than 0 or greater than 59.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="second"/> is less than 0 or greater than 59.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="millisecond"/> is less than 0 or greater than 999.
+        /// </exception>
+        /// <remarks>
+        /// This constructor interprets <paramref name="year"/>, <paramref name="month"/> and <paramref name="day"/> as a year, month and day
+        /// in the Gregorian calendar. To instantiate a <see cref="DateTime"/> value by using the year, month and day in another calendar, call
+        /// the <see cref="DateTime(int, int, int, int, int, int, int, int, Calendar)"/> constructor.
+        ///
+        /// The <see cref="Kind"/> property is initialized to <see cref="DateTimeKind.Unspecified"/>.
+        ///
+        /// For applications in which portability of date and time data or a limited degree of time zone awareness is important,
+        /// you can use the corresponding <see cref="DateTimeOffset"/> constructor.
+        /// </remarks>
         public DateTime(int year, int month, int day, int hour, int minute, int second, int millisecond)
+            : this(year, month, day, hour, minute, second, millisecond, 0)
```

Comment:
This will make all existing constructors pay extra cost of checking and calculating the microseconds ticks. I suggest avoiding that by not touching the old constructors and factoring this constructor code to its own method which we can call from the old and new constructor. We can mark the factored code method with aggressive inclining.